src/stdlib_linalg_state.fypp

```diff
@@ -0,0 +1,453 @@
+        integer,intent(in) :: flag
+        state_ge_flag = err%state >= flag
+     end function state_ge_flag
+
+     !> Compare an integer flag with the error state
+     elemental logical(lk) function flag_gt_state(flag,err)
+        integer,intent(in) :: flag
+        type(linalg_state),intent(in) :: err
+        flag_gt_state = err%state > flag
+     end function flag_gt_state
+
+     !> Compare an integer flag with the error state
+     elemental logical(lk) function flag_ge_state(flag,err)
+        integer,intent(in) :: flag
+        type(linalg_state),intent(in) :: err
+        flag_ge_state = err%state >= flag
+     end function flag_ge_state
+
+    !> Error creation message, with location location
+    pure type(linalg_state) function new_state(where_at,flag,a1,a2,a3,a4,a5,a6,a7,a8,a9,a10, &
+                                               v1,v2,v3,v4,v5)
+
+       !> Location
+       character(len=*),intent(in) :: where_at
+
+       !> Input error flag
+       integer,intent(in) :: flag
+
+       !> Optional scalar arguments
+       class(*),optional,intent(in) :: a1,a2,a3,a4,a5,a6,a7,a8,a9,a10
+
+       !> Optional vector arguments
+       class(*),optional,intent(in),dimension(:) :: v1,v2,v3,v4,v5
+
+       !> Create state with no message
+       new_state = new_state_nowhere(flag,a1,a2,a3,a4,a5,a6,a7,a8,a9,a10,v1,v2,v3,v4,v5)
+
+       !> Add location
+       if (len_trim(where_at) > 0) new_state%where_at = adjustl(where_at)
+
+    end function new_state
+
+    !> Error creation message, from N input variables (numeric or strings)
+    pure type(linalg_state) function new_state_nowhere(flag,a1,a2,a3,a4,a5,a6,a7,a8,a9,a10, &
+                                               v1,v2,v3,v4,v5) result(new_state)
+
+       !> Input error flag
+       integer,intent(in) :: flag
+
+       !> Optional scalar arguments
+       class(*),optional,intent(in) :: a1,a2,a3,a4,a5,a6,a7,a8,a9,a10
+
+       !> Optional vector arguments
+       class(*),optional,intent(in),dimension(:) :: v1,v2,v3,v4,v5
+
+       ! Init object
+       call new_state%destroy()
+
+       !> Set error flag
+       new_state%state = flag
+
+       !> Set chain
+       new_state%message = ""
+       call append(new_state%message,a1)
+       call append(new_state%message,a2)
+       call append(new_state%message,a3)
+       call append(new_state%message,a4)
+       call append(new_state%message,a5)
+       call append(new_state%message,a6)
+       call append(new_state%message,a7)
+       call append(new_state%message,a8)
+       call append(new_state%message,a9)
+       call append(new_state%message,a10)
+       call appendv(new_state%message,v1)
+       call appendv(new_state%message,v2)
+       call appendv(new_state%message,v3)
+       call appendv(new_state%message,v4)
+       call appendv(new_state%message,v5)
+
+    end function new_state_nowhere
+
+    ! Append a generic value to the error flag
+    pure subroutine append(msg,a,prefix)
```

Comment:
Exactly @certik, it covers all `real`, `integer`, `complex` kinds covered by stdlib (according to the preprocessing). I designed it this way so that error handling is easy to be read by the user but also easy to implement. See for example in my current `solve` implementation: 

https://github.com/perazz/fortran-lapack/blob/869e12f0aff09e3520a828e11e90898591872d47/src/stdlib_linalg_solve.f90#L178-L189

I find it very easy to read - and no need to write `format`s for each possible error string. Also, there are at least 5-6 possible error messages out of each LAPACK routine, it could be a `format` nightmare down the road.

Reply:
One option could be to move the numeric->string routines to `stdlib_io`?

Reply:
Could you use the `stdlib` [`to_string`](https://github.com/fortran-lang/stdlib/blob/e88daa336865b0a4a087b3ea2a15c1d1e94debc5/src/stdlib_strings.fypp#L14)?

Reply:
Totally, although here I purposedly avoided `allocatable` components, because there is going to be one `linalg_state_type` variable in every function, and I wanted the compiler to be able to put it on the stack to minimize performance penalty. What do you think?